src/pyhf/cli/patchset.py

```diff
@@ -0,0 +1,161 @@
+@click.argument('background-only', default='-')
+@click.argument('patchset', default='-')
+@click.option(
+    '--name', help='The name of the patch to extract.', default=None,
+)
+@click.option(
+    '--output-file',
+    help='The location of the output json file. If not specified, prints to screen.',
+    default=None,
+)
+def apply(background_only, patchset, name, output_file):
+    """
+    Apply a patch from patchset to the background-only workspace.
+
+    If the patchset does not contain the name, will exit with code 1.
+    If the patchset is not associated with the background-only workspace, will exit with code 2.
+
+    Returns:
+        ~pyhf.workspace.Workspace: The patched background-only workspace.
+    """
+    with click.open_file(background_only, 'r') as specstream:
+        spec = json.load(specstream)
+
+    utils.validate(spec, 'workspace.json')
+
+    with click.open_file(patchset, 'r') as fstream:
+        patchset_obj = json.load(fstream)
+
+    utils.validate(patchset_obj, 'patchset.json')
+
+    for hash_alg, digest in patchset_obj['metadata']['digests'].items():
+        digest_calc = utils.digest(spec, algorithm=hash_alg)
+        if not digest_calc == digest:
+            click.echo(
+                f"The digest verification failed for hash algorithm '{hash_alg}'.\n\tExpected: {digest}.\n\tGot:     {digest_calc}"
+            )
+            sys.exit(2)
+
+    extracted_patch_obj = None
+    for patch in patchset_obj['patches']:
+        if patch['name'] == name:
+            extracted_patch_obj = patch
+            break
+
+    if not extracted_patch_obj:
+        click.echo(f"There is no patch by the name '{name}' in this patchset.")
+        sys.exit(1)
+
+    result = jsonpatch.JsonPatch(extracted_patch_obj['patch']).apply(spec)
+
+    if output_file:
+        with open(output_file, 'w+') as out_file:
+            json.dump(result, out_file, indent=4, sort_keys=True)
+        log.debug("Written to {0:s}".format(output_file))
+    else:
+        click.echo(json.dumps(result, indent=4, sort_keys=True))
+
+
+@cli.command()
+@click.argument('background-only', default='-')
+@click.argument('patchset', default='-')
+def verify(background_only, patchset):
+    """
+    Verify the patchset digest against the background-only workspace.
+
+    If the patchset is not associated with the background-only workspace, will exit with code 2.
+
+    Returns:
+        ~pyhf.workspace.Workspace: The patched background-only workspace.
+    """
+    with click.open_file(background_only, 'r') as specstream:
+        spec = json.load(specstream)
+
+    utils.validate(spec, 'workspace.json')
+
+    with click.open_file(patchset, 'r') as fstream:
+        patchset_obj = json.load(fstream)
+
+    utils.validate(patchset_obj, 'patchset.json')
+
+    for hash_alg, digest in patchset_obj['metadata']['digests'].items():
+        digest_calc = utils.digest(spec, algorithm=hash_alg)
+        if not digest_calc == digest:
+            click.echo(
```

Comment:
Do you think it would be worth also printing out a confirmation message if the verification succeeds?